LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -430,20 +469,70 @@ public async Task RestoreLeaderboard_NotFound_WasNeverDeleted()
 
         Leaderboard board = new()
         {
-            Name = "Hyper Mario World",
+            Name = "Hyper Mario World Not Deleted",
             Slug = "hyper-mario-world-non-deleted",
         };
 
         context.Leaderboards.Add(board);
         await context.SaveChangesAsync();
         board.Id.Should().NotBe(default);
 
-        Func<Task<LeaderboardViewModel>> act = async () => await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/{board.Id}/restore", new()
+        try
         {
-            Jwt = _jwt
-        });
+            await _apiClient.Put<LeaderboardViewModel>(
+                $"/leaderboard/{board.Id}/restore",
+                new()
+                {
+                    Jwt = _jwt,
+                }
+            );
+        }
+        catch (RequestFailureException e)
+        {
+            e.Response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+            string? model = await e.Response.Content.ReadAsStringAsync();
+            model!.Should().MatchRegex("Not Deleted");
```

Comment:
Re: this bit: tried to call `ReadFromJsonAsync` instead, but the resolved object didn't have the message at all; it was just `null`. I can't figure out how to get the response to serialise right.